### DISCUSSION TITLE
Is there a built-in way to define and validate options as mutually exclusive?

### DISCUSSION THREAD

**Discussion by IanKemp** [spectreconsole/spectre.console#1291] [Q&A]
For example, CommandLineParser has the [`SetName` property on its `OptionAttribute` that can be used for this](https://github.com/commandlineparser/commandline/wiki/Mutually-Exclusive-Options).

It would be relatively trivial to implement this logic via `CommandSettings.Validate`, but I was wondering if there was a built-in way to  do this, that would format output in a default and standard way if multiple mutually exclusive options were present.

**Comment by nils-a** [comment]
Using `CommandSettings.Validate` is the built-way. 😬

There is no attribute to do this, however.
